### ISSUE TITLE
Scaled screenshots

### ISSUE THREAD

**Issue by OpenRift412** [dosbox-staging/dosbox-staging#718]
Another thought I had when switching between DOSBox Staging and Optionals is that because you can't screen-snip staging when it fullscreen, the only option you're given is the ability to take tiny native-res screenshots from games. 

A cool feature I'd love to see from Staging is the ability to take screenshots of DOSBox's fully processed output (with scaling/shaders added). Whether that be through allowing for screen-snipping in Windows or a built-in processed screenshot feature, it'd be nice to be able to take screenshots without having to go into windowed mode, or resize screenshots after the fact.

**Comment by kcgen** [comment]
Hmm, I wonder if this snippet will let us easily add this: https://stackoverflow.com/a/22339011

**Comment by dreamer** [comment]
> (…) because you can't screen-snip staging when it fullscreen (…)

What? I do it all the time on Linux. Normal OS screenshots do not work on Windows?!

Turbo Pascal 7.0 with crt-lottes-fast.glsl shader, as I am testing it for inclusion OOTB - I took the screenshot by pressing Alt+PrtScr (as I do for all applications on Gnome…).
![Screenshot from 2020-11-14 02-25-01](https://user-images.githubusercontent.com/3967/99135098-f66bc900-2620-11eb-9f47-55df6f2a03c0.png)

**Comment by dreamer** [comment]
Just tested and indeed… there's no way to make a fullscreen screenshot of dosbox-staging window on Windows…

I tried using both built-in Windows functionality (didn't work, it made the screenshot while I was leaving the window) and screenshot functionality built-in my Radeon drivers (didn't work, dosbox-staging was taking over the input event).

What's the behaviour on macOS?

**Comment by kcgen** [comment]
https://getgreenshot.org/

Back when I used a Windows-based work PC, that was one of my must-have open-source tools. 

After installing, configure its actions to save to png and avoid prompting. You'll have hotkeys for all the usual types of shots: fullscreen capture, active-window only, or selected rectangle. 

I know it's a workaround in this case.. but it's a very nice tool in general while we assess the way forward.  (Sad to see Microsoft has crippled their out-of-the-box experience for users!)

**Comment by arrowgent** [comment]
which render is being used?  surface, opengl, etc?

screencap works fine in linux (mate/gnome2-fork) when pressing PRTSCR

**Comment by dreamer** [comment]
My testing indicates all renderers/outputs have problem with screenshot in fullscreen on Windows… I suspect it's similar problem to the emulator pausing while the window is being moved or resized (also Windows-only bug, caused by the way we create/handle window).

**Comment by kcgen** [comment]
Some screenshot tools can't access the accelerated content, and you're left with an empty frame.

**Comment by kcgen** [comment]
Implemented in # https://github.com/dosbox-staging/dosbox-staging/pull/727

@OpenRift412 , Windows binaries here: https://github.com/dosbox-staging/dosbox-staging/actions/runs/375381819

Or build branch: `kc/display-capture-1`

I've set `Ctrl + F3` as the capture key for now. It's name is **"CapDisplay"** in the key-mapper, so feel free to remap as well.

This is just to let us try it out. We would of-course put more thought into its final hotkey and name, and how it relates to the existing (broken-as-named, IMO) screenshot feature.  

PS- regarding my option in that the current screenshot feature is broken: I do feel the existing screenshot is perfectly adequate if it were named "raw capture" or "DOS capture".
